server/controllers/deviceWearerResponsibleAdultController.test.ts

```diff
@@ -0,0 +1,268 @@
+        staffId: 123,
+      },
+    },
+    redirect: jest.fn(),
+    render: jest.fn(),
+    set: jest.fn(),
+    send: jest.fn(),
+  }
+}
+
+const createMockOrder = (status: OrderStatus, responsibleAdultFullName: string): Order => {
+  return {
+    id: uuidv4(),
+    status,
+    deviceWearer: {
+      nomisId: null,
+      pncId: null,
+      deliusId: null,
+      prisonNumber: null,
+      firstName: 'tester',
+      lastName: 'testington',
+      alias: 'test',
+      dateOfBirth: '2010-01-01T00:00:00.000Z',
+      adultAtTimeOfInstallation: false,
+      sex: 'male',
+      gender: 'male',
+      disabilities: ['Vision', 'Mobilitiy'],
+    },
+    deviceWearerContactDetails: {
+      contactNumber: '',
+    },
+    deviceWearerAddresses: [],
+    additionalDocuments: [],
+    deviceWearerResponsibleAdult: {
+      relationship: 'parent',
+      otherRelationshipDetails: null,
+      fullName: responsibleAdultFullName,
+      contactNumber: 'parent contact',
+    },
+  }
+}
+
+describe('DeviceWearerController', () => {
+  let mockRestClient: jest.Mocked<RestClient>
+  let mockAuditClient: jest.Mocked<HmppsAuditClient>
+  let mockAuditService: jest.Mocked<AuditService>
+  let mockDeviceWearerResponsibleAdultService: jest.Mocked<DeviceWearerResponsibleAdultService>
+  let deviceWearerResponsibleAdultController: DeviceWearerResponsibleAdultController
+
+  beforeEach(() => {
+    mockAuditClient = new HmppsAuditClient({
+      queueUrl: '',
+      enabled: true,
+      region: '',
+      serviceName: '',
+    }) as jest.Mocked<HmppsAuditClient>
+    mockRestClient = new RestClient('cemoApi', {
+      url: '',
+      timeout: { response: 0, deadline: 0 },
+      agent: { timeout: 0 },
+    }) as jest.Mocked<RestClient>
+    mockAuditService = new AuditService(mockAuditClient) as jest.Mocked<AuditService>
+    mockDeviceWearerResponsibleAdultService = new DeviceWearerResponsibleAdultService(
+      mockRestClient,
+    ) as jest.Mocked<DeviceWearerResponsibleAdultService>
+    deviceWearerResponsibleAdultController = new DeviceWearerResponsibleAdultController(
+      mockAuditService,
+      mockDeviceWearerResponsibleAdultService,
+    )
+  })
+
+  describe('view', () => {
+    it('should render the form using the saved responsible adult data', async () => {
+      // Given
+      const mockOrder = createMockOrder(OrderStatusEnum.Enum.IN_PROGRESS, 'Parent Name')
+      const req = createMockRequest(mockOrder)
+      const res = createMockResponse()
+      const next = jest.fn()
+      req.flash = jest.fn().mockReturnValue([])
+
+      // When
+      await deviceWearerResponsibleAdultController.view(req, res, next)
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith(
+        'pages/order/about-the-device-wearer/responsible-adult-details',
+        expect.objectContaining({
+          orderSummaryUri: '/order/123456789/summary',
+          formActionUri: '/order/123456789/about-the-device-wearer/responsible-adult',
```

Comment:
This shouldn't be needed as the `post` route is the same as the page we're on, will need removing in the `njk` template too